src/components/DisciplineListItem.tsx

```diff
@@ -10,27 +16,66 @@ interface DisciplineListItemProps {
   hasBadge: boolean
   rightChildren?: ReactElement
   disabled?: boolean
+  showProgress?: boolean
 }
 
+const Icon = styled.Image`
+  width: ${wp('7%')}px;
+  height: ${wp('7%')}px;
+`
+
+const IconContainer = styled.View`
+  position: absolute;
+  top: ${wp('3.5%')}px;
+  left: ${wp('3.5%')}px;
+`
+
 const DisciplineListItem = ({
   item,
   onPress,
   hasBadge,
   rightChildren,
-  disabled = false
+  disabled = false,
+  showProgress = false
 }: DisciplineListItemProps): ReactElement | null => {
   const { numberOfChildren, title, icon, apiKey } = item
   const badgeLabel = hasBadge ? numberOfChildren.toString() : undefined
   // Description either contains the number of children and the type of children or just the type of children if the number is shown as badge
   const description = hasBadge ? childrenLabel(item) : childrenDescription(item)
 
+  const [progress, setProgress] = useState<number>(0)
+
+  useFocusEffect(() => {
+    if (showProgress) {
+      getProgress(item).then(setProgress).catch(reportError)
+    }
+  })
+
+  const iconWithProgress = (
+    <>
+      <Progress.Circle
+        progress={progress}
```

Comment:
```suggestion
  const { data: progress } = useReadProgress(item)

  const iconWithProgress = (
    <>
      <Progress.Circle
        progress={progress ?? 0}
```

Reply:
I think this does not really make sense, as we would make an slow reading request, also if it is not needed, because progress is not displayed.

Reply:
somehow missed the if clause, nvm :)

Reply:
Well it is reading from memory, but "slow" is relative. But there are multiple requests because the screen render multiple DisicplineListItems, therefore I would prefer to stick to not load if not needed, although it wouldn't be a big problem to do so.